tests/scripts/test_detached_actor_2.py

```diff
@@ -0,0 +1,27 @@
+import ray
+import time
+import sys
+
+ray.init(address='ray://127.0.0.1:10001', namespace=sys.argv[1])
+
+def retry_with_timeout(func, timeout=90):
+    err = None
+    for _ in range(timeout):
+        try:
+            return func()
+        except BaseException as e:
```

Comment:
Why BaseException vs Exception?

Reply:
https://docs.python.org/3.10/library/exceptions.html#exception-hierarchy

`BaseException` is at the top of the exception hierarchy. `Exception` does not include `SystemExit`, `KeyboardInterrupt`, and `GeneratorExit`. `BasicException` can catch all exceptions, but also catch the exception when you want to kill this process. I was not sure which kind of exception will be thrown here, and thus I used `BasicException`.

Reply:
Makes sense, thanks for explaining!